EPRN.Portal/Services/PRNService.cs

```diff
@@ -224,11 +225,25 @@ public async Task<ViewPRNViewModel> GetViewPrnViewModel(string reference)
 
         public async Task<ActionPrnViewModel> GetActionPrnViewModel(int id)
         {
-            //TODO: This will need to be populated by the journey so we can return to it.
-            return new ActionPrnViewModel
-            {
-                Id = id
-            };
+            var reference = await _httpPrnsService.GetPrnReference(id);
+            var dto = await _httpPrnsService.GetPrnDetails(reference);
+
+            var actionPrnViewModel = new ActionPrnViewModel { Id = id };
```

Comment:
Previous code is not used! Is this unfinished work? If so you need to double check over your work before checking it in. Ensure that it works from all aspects. Have you written unit tests? Because if you have they should be failing, and if they're not, then they are incorrect. If you have not written any, you should do. Every controller and service needs unit tests, even for the most simple bit of code.

Reply:
Done.